modulemd_tools/modulemd_tools/yaml.py

```diff
@@ -0,0 +1,300 @@
+"""
+Module for working with modulemd YAML definitions with the least abstractions as
+possible. Within this module, modulemd YAMLs are represented simply as a string,
+and all transformation functions are `str` -> `str`.
+"""
+
+import os
+import gi
+import yaml
+gi.require_version("Modulemd", "2.0")
+from gi.repository import Modulemd
+
+
+def is_valid(mod_yaml):
+    """
+    Determine whether the `mod_yaml` string is a valid modulemd YAML definition
+    """
+    idx = Modulemd.ModuleIndex.new()
+    ret, _ = idx.update_from_string(mod_yaml, strict=True)
```

Comment:
This may throw an exception if the YAML itself is invalid and unparseable. (For example, it starts a map or array but never terminates it).